handlers/controller.js

```diff
@@ -157,6 +122,68 @@ let addProductByLookup = async(req, res) => {
     });
 };
 
+
+/**
+ * Returns the top five manufacturers of the same category of product
+ * @param req request 
+ * @param res  response
+ */
+let getTopManufacturers = async(req, res) => {
+    connectDb().then(async () => {
+        if(!req.params.id) {
+            return res.status(StatusCode.PRECONDITION_FAILED).send(null);
+        }      
+        let barcode = req.params.id;
+        try {
+            let doc = await models.Product.findOne({ barcode });
+            if(doc) {
+                console.log(`found ${doc.barcode} in mongodb`);
+                if(!doc.category) {
+                    console.error(`category not found for ${doc.name}`)
+                    return res.status(StatusCode.PRECONDITION_FAILED).send();
```

Comment:
Can we change this status code?
```suggestion
return res.status(StatusCode.NOT_FOUND).send();
```